src/routes/v2/authenticated/review.ts

```diff
@@ -41,25 +42,50 @@ export class ReviewsRouter {
 
   private readonly notificationsService
 
+  private readonly githubService
+
   constructor(
     reviewRequestService: ReviewRequestService,
     identityUsersService: UsersService,
     sitesService: SitesService,
     collaboratorsService: CollaboratorsService,
-    notificationsService: NotificationsService
+    notificationsService: NotificationsService,
+    githubService: GitHubService
   ) {
     this.reviewRequestService = reviewRequestService
     this.identityUsersService = identityUsersService
     this.sitesService = sitesService
     this.collaboratorsService = collaboratorsService
     this.notificationsService = notificationsService
+    this.githubService = githubService
 
     autoBind(this)
   }
 
+  // NOTE: This is required because we want to distinguish
+  // between unmigrated sites that don't exist in our db
+  // vs sites that truly don't exist.
+  // This is to avoid flooding the frontend with 404s
+  // and triggering false alarms
+  // TODO (IS-58): Remove this function and related calls when
+  // all sites are migrated over to our db
+  doesUnmigratedSiteExist = async (
+    sessionData: UserWithSiteSessionData
+  ): Promise<boolean> => {
+    try {
+      // Site exists on github but isn't migrated
+      await this.githubService.getRepoInfo(sessionData)
+      return true
+    } catch {
+      // If there are any errors, we failed to fetch the site
+      // and the site is assumed to not exist.
+      return false
```

Comment:
in the event site does not exist, what response does the github API return?

Also, when we say errors, do we mean any response apart from 2xx goes into the catch block?

Reply:
returns 404 on our side - this means that it falls through to the default case where the site really doesn't exist

re: second point, yes. broadly, if it's a 4xx, we can omit the response from github because it's user based so our subsequent flow should cover it. if it's truly github's fault (5xx), we can't determine if the site exists or not, so returning that it **doesn't** is a safer choice than saying it does (and possibly being wrong)

Reply:
okay makes sense